pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -162,9 +163,45 @@ func (h *upgradeHandler) OnChanged(_ string, upgrade *harvesterv1.Upgrade) (*har
 			toUpdate.Status.UpgradeLog = upgradeLog.Name
 		}
 		harvesterv1.LogReady.CreateUnknownIfNotExists(toUpdate)
+		harvesterv1.LogReady.LastUpdated(toUpdate, time.Now().UTC().Format(time.RFC3339))
 		return h.upgradeClient.Update(toUpdate)
 	}
 
+	if harvesterv1.LogReady.IsUnknown(upgrade) {
```

Comment:
@dharmit  Thanks.

Could you put this new block of code into an independent function? the current `OnChanged` is too big and needs to be splited step by step.

And, could `upgradeLogCondLogReadTimeout` be defined from the https://docs.harvesterhci.io/v1.5/advanced/index#upgrade-config, also with a valid range check like [1..20] minutes.

Reply:
Thanks for the review. Pushed changes.